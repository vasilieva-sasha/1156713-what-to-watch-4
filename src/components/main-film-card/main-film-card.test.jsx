import React from "react";
import renderer from "react-test-renderer";
import MainFilmCard from './main-film-card';
import configureStore from 'redux-mock-store';
import NameSpace from './../../reducer/name-space';
import {Provider} from "react-redux";

const mock = {
  title: `title`,
  genre: `genre`,
  releaseDate: 0,
  background: ``,
  posterInfo: ``
};

const mockStore = configureStore([]);

it(`Main film card renders correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      promoFilm: mock,
    },
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`,
    },
  });
  const tree = renderer.create(
      <Provider store={store}>
        <MainFilmCard onPlayClick={() => {}}/>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
