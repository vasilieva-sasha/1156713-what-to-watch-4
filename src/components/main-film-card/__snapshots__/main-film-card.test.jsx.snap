// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main film card renders correctly 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="title"
      src=""
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="#"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="title poster"
          height="327"
          src=""
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          title
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            genre
          </span>
          <span
            className="movie-card__year"
          >
            0
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/films/1/player"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
